var/spack/repos/builtin/packages/py-adios/package.py

```diff
@@ -0,0 +1,72 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import inspect
+
+
+class PyAdios(PythonPackage):
+    """NumPy bindings of ADIOS1"""
+
+    homepage = "https://www.olcf.ornl.gov/center-projects/adios/"
+    url      = "https://github.com/ornladios/ADIOS/archive/v1.12.0.tar.gz"
+
+    version('develop', git='https://github.com/ornladios/ADIOS.git',
+            branch='master')
+    version('1.12.0', '84a1c71b6698009224f6f748c5257fc9')
+    #version('1.11.1', '5639bfc235e50bf17ba9dafb14ea4185')
+    #version('1.11.0', '5eead5b2ccf962f5e6d5f254d29d5238')
+    #version('1.10.0', 'eff450a4c0130479417cfd63186957f3')
+    #version('1.9.0', '310ff02388bbaa2b1c1710ee970b5678')
+
+    variant('mpi', default=True,
+            description='Enable MPI support')
+
+    #depends_on('adios@{0}'.format(self.spec.version), type=['build', 'link', 'run'])
```

Comment:
I would like to express here, that `py-adios@1.12.0` depends on `adios@1.12.0`.
An other version depends on its version again.

Is there a generic way to query the version of *this* package at that location or do I need to write a cascade of `when` clauses for each version?

Reply:
There is currently no way to do this, although I wish there was. You could write a for-loop and loop over each version.

Reply:
great idea, thank you. I always forget how flexible python classes are!

Reply:
ah, but `self.versions.values()` does not exist at this place yet :'(

Reply:
uh, wait. how?
`self.versions` does not exist at this place yet.

Reply:
What I'm saying is that if you know you have 5 versions, you could do something like:
```python
for version in ['1.10.0', '1.10.1', '1.11.0', '1.11.2', '1.12.0']:
    depends_on('adios@{0}'.format(version), when='@{0}'.format(version))
```
instead of having to write everything out 5 times. A little better. But I agree, it would be nice to say "depends on the same version". It is a fairly common constraint in packages like this.

Reply:
ah, of course. I was forgetting about the `when='@{0}'` part! Thanks!